api/v1alpha1/ngrok_common.go

```diff
@@ -95,3 +98,29 @@ type EndpointCircuitBreaker struct {
 	// Error threshold percentage should be between 0 - 1.0, not 0-100.0
 	ErrorThresholdPercentage resource.Quantity `json:"errorThresholdPercentage,omitempty"`
 }
+
+type EndpointOIDC struct {
+	// Do not enforce authentication on HTTP OPTIONS requests. necessary if you are
+	// supporting CORS.
+	OptionsPassthrough bool `json:"optionsPassthrough,omitempty"`
+	// the prefix of the session cookie that ngrok sets on the http client to cache
+	// authentication. default is 'ngrok.'
+	CookiePrefix string `json:"cookiePrefix,omitempty"`
+	// Duration of inactivity after which if the user has not accessed
+	// the endpoint, their session will time out and they will be forced to
+	// reauthenticate.
+	//+kubebuilder:validation:Format=duration
+	InactivityTimeout v1.Duration `json:"inactivityTimeout,omitempty"`
```

Comment:
I'm thinking we might want to standardize on using `v1.Duration` across the board for ngrok api fields that represent durations so we can just do `inactivityTimeout: 4h` instead of `inactivityTimeout: 14400`. We can do the conversion to the types necessary for ngrok-go

Reply:
👍 having the unit feels like the k8s way